### PR TITLE
build: transpile triggers with ts-node api

### PR DESCRIPTION
**Pull request by trim21**
remove a TODO in `util/process_triggers_folder.ts`

>  // Generate javascript from typescript.
>  // TODO: replace this with programatic use of TypeScript API.

not extractly using typescript, but calling ts-node's transpile api, so type checking is also removed ( still covered by other tsc ).

This PR also make this script able to be executed in any cwd.

### CODE REVIEW

File: util/process_triggers_folder.ts [hunk 1]
```diff
@@ -109,16 +115,30 @@ const lint = async (filename: string, lines: string[]) => {
...
-  const originalFilename = outputFileToOrigFile(filename);
-  let lines = originalContents.split(/[\r\n]+/);
+const tsc = tsNode.create({
+  transpileOnly: true,
+  project: path.join(__dirname, '../tsconfig.json'),
+  // compilerOptions: {
+  //   target: 'ES2020',
+  //   declaration: false,
+  //   declarationMap: false,
+  // },
```
Comment (quisquous) [inline util/process_triggers_folder.ts:125]:
```suggestion
```

Should this comment be removed?

File: util/process_triggers_folder.ts [hunk 2]
```diff
@@ -159,51 +179,40 @@ const processFile = async (filename: string) => {
...
   // Overwrite the file.
-  fs.writeFileSync(filename, contents);
+  fs.mkdirSync(path.dirname(distFilePath), { recursive: true });
+  fs.writeFileSync(distFilePath.replace('.ts', '.js'), contents);
 
   // Copy timeline file if present
-  const importPath = path.join(
-    '..',
-    originalFilename.replace(path.extname(originalFilename), '.ts'),
+  const importPath = ('../ui/raidboss/data/' + path.relative(root, originalFilename)).replace(
```
Comment (quisquous) [inline util/process_triggers_folder.ts:192]: Thanks for fixing this.  I have no idea how this ever worked locally without the slash replacement.

Comment (trim21) [reply util/process_triggers_folder.ts:192]: yes, it didn't work...

**Comment by trim21** [comment]
It looks like eslint is not necessary here? It can also to be removed.

**Comment by quisquous** [comment]
> It looks like eslint is not necessary here? It can also to be removed.

I'm not sure.  I felt like it was there for a reason, but I'd have to look at a diff to believe it.  I think this is fine to land and eslint can be removed separately.

**Comment by trim21** [comment]
> > It looks like eslint is not necessary here? It can also to be removed.
> 
> I'm not sure. I felt like it was there for a reason, but I'd have to look at a diff to believe it. I think this is fine to land and eslint can be removed separately.

https://github.com/trim21/temp-repo-show-diff/commit/a0520596737a6d1a93457fada9ebd46de2145e43

**Comment by quisquous** [comment]
Yeah, looks like it's there for a reason, then.  I'd prefer that the transpiled triggers look as close to the original as possible.

**Pull request merged**